developer/src/kmcmpdll/CheckFilenameConsistency.cpp

```diff
@@ -53,51 +58,71 @@ BOOL IsRelativePath(wchar_t const * p) {
   //  C:\...\BITMAP.BMP
   //  \\SERVER\SHARE\...\BITMAP.BMP
 
+#if defined(_WIN32) || defined(_WIN64)
   if (*p == L'\\') return FALSE;
+#else
+  if (*p == L'/') return FALSE;
+#endif
+
   if (*p && *(p + 1) == L':') return FALSE;
 
   return TRUE;
 }
 
-DWORD CheckFilenameConsistency(char const * Filename, BOOL ReportMissingFile) {
-  PWCHAR WFilename = strtowstr((char *)Filename);
-  DWORD const result = CheckFilenameConsistency(WFilename, ReportMissingFile);
-  delete WFilename;
+KMX_DWORD CheckFilenameConsistency(char const * Filename, BOOL ReportMissingFile) {
+  PKMX_WCHAR WFilename = strtowstr((char *)Filename);
+  KMX_DWORD const result = CheckFilenameConsistency(u16fmt(WFilename).c_str(), ReportMissingFile);
+  delete WFilename;  
   return result;
 }
 
-DWORD CheckFilenameConsistency(wchar_t const * Filename, BOOL ReportMissingFile) {
-  WCHAR Name[_MAX_PATH], FName[_MAX_FNAME], Ext[_MAX_EXT];
-  _wfinddata_t fi;
+KMX_DWORD CheckFilenameConsistency(KMX_WCHART const * Filename, bool ReportMissingFile) {
+  KMX_WCHART Name[_MAX_PATH], FName[_MAX_FNAME], Ext[_MAX_EXT];
+  KMX_WCHAR ErrExtra[256];
   intptr_t n;
 
   if (IsRelativePath(Filename)) {
-    PWCHAR WCompileDir = strtowstr(CompileDir);
-    wcscpy_s(Name, _countof(Name), WCompileDir);  // I3481
+    PKMX_WCHAR WCompileDir = strtowstr(CompileDir);
+    wcscpy_s(Name, _countof(Name), u16fmt(WCompileDir).c_str());  // I3481
     wcscat_s(Name, _countof(Name), Filename);  // I3481
   }
   else {
     wcscpy_s(Name, _countof(Name), Filename);  // I3481
   }
 
-  if ((n = _wfindfirst(Name, &fi)) == -1) {
+#if defined(_WIN32) || defined(_WIN64)
+  _wfinddata_t fi;
+  n = _wfindfirst(Name, &fi);
+#else
+  n= access(Name,F_OK);
+#endif
+
+  if (n == -1){
     if (ReportMissingFile) {
-      wsprintf(ErrExtra, "referenced file '%ls'", Filename);
+      u16sprintf(ErrExtra,_countof(ErrExtra),L"referenced file %ls",Filename);
       AddWarning(CWARN_MissingFile);
     }
     return CERR_None;
   }
 
```

Comment:
Is not using _findclose()   in the original code done on purpose?

Reply:
That's probably a bug, I'd say

Reply:
👍